nmdc_schema/migrators/partials/migrator_from_11_1_0_to_11_2_0/migrator_from_11_1_0_to_11_2_0_part_1.py

```diff
@@ -0,0 +1,246 @@
+        2
+        """
+
+        # Make handy aliases for some values.
+        data_generation_id = data_generation["id"]
+        data_generation_analyte_category = data_generation.get("analyte_category")
+
+        # Implement the algorithm discussed with schema maintainers.
+        if data_generation_analyte_category == "nom" or data_generation_analyte_category == "metabolome":
+
+            # Determine which calibration-related field this document has, if any.
+            calibration_field_name = None
+            if "generates_calibration" in data_generation:
+                calibration_field_name = "generates_calibration"
+            elif "has_calibration" in data_generation:
+                calibration_field_name = "has_calibration"
+            else:
+                self.logger.warning(f"'data_generation_set' document '{data_generation_id}' has neither field")
+
+            if calibration_field_name is not None:
+
+                # Find the `workflow_execution_set` document whose `was_informed_by` field contains this document's `id`
+                workflow_execution = self.adapter.get_document_having_value_in_field(
+                    collection_name="workflow_execution_set",
+                    field_name="was_informed_by",  # confirmed: this is a single valued field, not a multivalued one
+                    value=data_generation_id,
+                )
+                if workflow_execution is not None:
+
+                    # Update the instance-level dictionary, which will be "consumed" by a different method.
+                    self.calibration_mappings[workflow_execution["id"]] = data_generation.get(calibration_field_name)
+
+                    # If this `data_generation_set` document's `analyte_category` value was "nom", specifically,
+                    # delete the calibration-related field from the document.
+                    if data_generation_analyte_category == "nom":
+                        data_generation.pop(calibration_field_name)
+                else:
+                    self.logger.warning(f"No 'workflow_execution_set' document has a "
+                                        f"'was_informed_by' value of '{data_generation_id}'")
+
+        return data_generation
+
+    def apply_calibration_mapping(self, workflow_execution: dict) -> dict:
+        r"""
+        Sets the `uses_calibration` field to the value defined in the instance-level dictionary.
+
+        >>> from nmdc_schema.migrators.adapters.dictionary_adapter import DictionaryAdapter
+        >>> m = Migrator(adapter=DictionaryAdapter(database={}))
+        >>> m.calibration_mappings = {
+        ...     'nmdc:wfnom-99-abc123': 'nmdc:calib-99-def456',
+        ...     'nmdc:wfnom-99-foobar': 'nmdc:calib-99-bazqux',
+        ... }
+        >>> m.apply_calibration_mapping({'id': 'nmdc:wfnom-99-abc123', 'type': 'nmdc:MetagenomeAnnotation'})  # id is in mapping dictionary
```

Comment:
minor but the typecode and the type are inconsisent here, making the example confusing.

Reply:
Thanks, @aclum. I saw your overall approval comment. I'll try to address this real quick in an attempt to make the example less confusing.